### DISCUSSION TITLE
Cannot see Starburst or Trino in Metabase UI

### DISCUSSION THREAD

**Discussion by hongbo-miao** [starburstdata/metabase-driver#130] [Q&A] [answered]
I tried to base on this tutorial to set up: https://docs.starburst.io/clients/metabase.html#requirements

I am using

- Docker image: [metabase/metabase v0.49.13](https://hub.docker.com/layers/metabase/metabase/v0.49.13/images/sha256-73139ef44d6b1e35321655fb6f5869f8377047763d514ed050298d0bdc785959?context=explore) (latest as of today)
- metabase-driver 4.1.0

Here is how I run:

```sh
docker run --publish=3000:3000 \
	--mount type=bind,source="$(pwd)/plugins",destination=/plugins \
	--name metabase \
	metabase/metabase:v0.49.13
```

I can see the jar file has successfully mounted into container's `/plugins` folder:

![image](https://github.com/starburstdata/metabase-driver/assets/3375461/09711d75-3960-48a9-b06e-9c78f2d1023b)

But the Metabase UI does not show Starburst or Trino:

![image](https://github.com/starburstdata/metabase-driver/assets/3375461/6fecc577-4c89-49df-acec-68293710e7b1)

![image](https://github.com/starburstdata/metabase-driver/assets/3375461/fb4a83ae-3b33-4c2c-aa80-21e381d3adfb)

I did see open source Metabase supports third-party driver https://www.metabase.com/docs/latest/developers-guide/partner-and-community-drivers

![image](https://github.com/starburstdata/metabase-driver/assets/3375461/48041b66-baca-44db-b9f2-fd59d513c212)

I saw someone asked a [similar question](https://stackoverflow.com/questions/75757171/add-trino-driver-to-metabase-in-docker-does-not-see-trino-source-in-metabase-w) a while ago on Stack Overflow as well.

Any ideas? Thanks! ☺️

---

# UPDATE 1

Hmm I saw this Metabase driver supported Metabase version v1.48.3 is a Metabase Enterprise version at https://github.com/starburstdata/metabase-driver/blob/main/app_versions.json

I tried Metabase open source v0.48.3 (I guess it corresponds Metabase Enterprise v1.48.3 ?) and also an even elder version v0.47.0, but still same, not showing Trino.

Does it mean it does not support open source Metabase? If supports, which version is supported? Thanks!

**Comment by lpoulain** [comment]
I tried to reproduce the issue locally but I am encountering some technical difficulties. In the meantime, could you try renaming the driver into `starburst.metabase-driver.jar`? Also, do you see any message in the logs around the plugins?

**Reply by hongbo-miao** [reply]
Thanks @lpoulain ! I have renamed to `starburst.metabase-driver.jar`, but still same

![image](https://github.com/starburstdata/metabase-driver/assets/3375461/0f6dd3fa-7da5-4082-9b5e-1de7532c6fdd)

If your technical issue is running on MacBook M chip, you can use this command with `bobblybook/metabase` image instead ☺️

```sh
docker run --publish=3000:3000 \
	--mount type=bind,source="$(pwd)/plugins",destination=/plugins \
	--name metabase \
	bobblybook/metabase:v0.49.13
```

**Comment by lpoulain** [comment, marked as answer]
Indeed I can start it in Docker with this image. The loading of the plugins seems to be different than on the enterprise version (looking at the logs it doesn't seem to look at what's in `plugins` but `modules`. I was able to get the Starburst driver recognized by changing the binding (`source="$(pwd)/plugins",destination=/app/plugins`), and get it working. But you can also create your Dockerfile.

But the bottom line is that the Starburst driver does work with the free version of Metabase.

**Reply by hongbo-miao** [reply]
Wow, works perfect now! Thank you so much! ☺️
Just summarize the final solutions:

Prepare

```sh
mkdir plugins
curl --silent --fail --show-error --location --output plugins/starburst-4.1.0.metabase-driver.jar https://github.com/starburstdata/metabase-driver/releases/download/4.1.0/starburst-4.1.0.metabase-driver.jar
```

x86 chips:

```sh
docker run --publish=3000:3000 \
	--volume="$(pwd)/plugins:/app/plugins" \
	--name=metabase \
	metabase/metabase:v0.49.13
```

Apple M-series chips:
```sh
docker run --publish=3000:3000 \
	--volume="$(pwd)/plugins:/app/plugins" \
	--name=metabase \
	bobblybook/metabase:v0.49.13
```
